modules/effects/arrow-effects-data/src/main/kotlin/arrow/effects/IO.kt

```diff
@@ -123,6 +126,12 @@ sealed class IO<out A> : IOOf<A> {
     /* For parMap, look into IOParallel */
   }
 
+  suspend fun suspended(): A = suspendCoroutine { cont ->
```

Comment:
`await` fits better here

Reply:
Wouldn't it be confused with `async/await`?

Reply:
Originally we had it as `val suspended` but that felt less ergonomic to use since you can just do `io::suspended` instead but not the other way around

Reply:
Maybe it collides with `await`, although it's semantically the same. In any case I'd like a verb as a name.

Reply:
I liked `val suspended: suspend () -> A` but that requires excessive wrapping, which I wanted to avoid this way.

The semantics with `Deferred#await` are not the same (well..), in Kotlinx they always talk about `await`ing eager async operations. That's why I am not a big fan of `await` but I agree a verb would be better.